driver-examples/osgi/src/test/java/com/datastax/driver/osgi/MailboxServiceTest.java

```diff
@@ -0,0 +1,173 @@
+
+    private CompositeOption defaultOptions = new CompositeOption() {
+
+        @Override public Option[] getOptions() {
+            return options(
+                systemProperty("cassandra.contactpoints").value(CCMBridge.IP_PREFIX + 1),
+                bundle("reference:file:target/classes"),
+                mavenBundle("com.codahale.metrics", "metrics-core", "3.0.2"),
+                mavenBundle("org.slf4j", "slf4j-api", "1.7.5"),
+                mavenBundle("org.slf4j", "slf4j-simple", "1.7.5").noStart(),
+                systemPackages("org.testng", "org.junit", "org.junit.runner", "org.junit.runner.manipulation",
+                    "org.junit.runner.notification", "com.jcabi.manifests")
+            );
+        }
+    };
+
+    @Configuration
+    public Option[] shadedConfig() {
+        return options(
+            driverBundle.classifier("shaded"),
+            guavaBundle,
+            defaultOptions
+        );
+    }
+
+    @Configuration
+    public Option[] defaultConfig() {
+        return options(
+            driverBundle,
+            guavaBundle,
+            nettyBundle,
+            defaultOptions
+        );
+    }
+
+    @Configuration
+    public Option[] guava15Config() {
+        return options(
+            driverBundle,
+            nettyBundle,
+            guavaBundle.version("15.0"),
+            defaultOptions
+        );
+    }
+
+    @Configuration
+    public Option[] guava16Config() {
+        return options(
+            driverBundle,
+            nettyBundle,
+            guavaBundle.version("16.0.1"),
+            defaultOptions
+        );
+    }
+
+    @Configuration
+    public Option[] guava17Config() {
+        return options(
+            driverBundle,
+            nettyBundle,
+            guavaBundle.version("17.0"),
+            defaultOptions
+        );
+    }
+
+    @Configuration
+    public Option[] guava18Config() {
+        return options(
+            driverBundle,
+            nettyBundle,
+            guavaBundle.version("18.0"),
+            defaultOptions
+        );
+    }
+
+    /**
+     * <p>
+     * Exercises a 'mailbox' service provided by an OSGi bundle that depends on the driver.  Ensures that
+     * queries can be made through the service with the current given configuration.
+     * </p>
+     *
+     * The following configurations are tried (defined via methods with the @Configuration annotation):
+     * <ol>
+     *   <li>Default bundle (Driver with all of it's dependencies)</li>
+     *   <li>Shaded bundle (Driver with netty shaded)</li>
+     *   <li>With Guava 15</li>
+     *   <li>With Guava 16</li>
+     *   <li>With Guava 17</li>
+     *   <li>With Guava 18</li>
+     * </ol>
+     *
+     * @test_category packaging
+     * @expected_result Can create, retrieve and delete data using the mailbox service.
+     * @jira_ticket JAVA-620
+     * @since 2.0.10, 2.1.5
+     */
+    public void service_api_functional() throws MailboxException {
```

Comment:
Pretty basic test, but ensures that the driver is loaded properly and can interface with a cassandra cluster.